src/CPUSolver.cpp

```diff
@@ -819,14 +819,36 @@ void CPUSolver::transferBoundaryFlux(int track_id,
   /* For the "forward" direction */
   if (direction) {
     start = _tracks[track_id]->isNextOut() * _polar_times_groups;
-    bc = std::min((int)_tracks[track_id]->getBCOut(), 1);
+
+    /* If the outgoing boundary condition is vacuum, set bc to 0 indicating the
+     * track flux of the outgoing track will be zeroed out */
```

Comment:
Ok, I'll have to change the Track's `_bc_in` and `_bc_out` attributes to int types then. What do you think about changing these attributes to bool types named something more like `_transfer_flux_out` and `_transfer_flux_in` and then casting them as ints in the solver?